framework/src/timeintegrators/ActuallyExplicitEuler.C

```diff
@@ -105,85 +246,161 @@ ActuallyExplicitEuler::preSolve()
+
+    u_dotdot += *_solution;
+    u_dotdot -= _solution_old;
+    u_dotdot -= _solution_old;
+
+    u_dotdot *= 1.0 / _dt / _dt;
+    u_dotdot.close();
+  }
 }
 
 void
 ActuallyExplicitEuler::solve()
 {
   auto & es = _fe_problem.es();
-
   auto & nonlinear_system = _fe_problem.getNonlinearSystemBase();
-
   auto & libmesh_system = dynamic_cast<NonlinearImplicitSystem &>(nonlinear_system.system());
-
-  auto & mass_matrix = *libmesh_system.matrix;
-
-  _current_time = _fe_problem.time();
+  auto & system_matrix = *libmesh_system.matrix;
 
   // Set time back so that we're evaluating the interior residual at the old time
+  _current_time = _fe_problem.time();
   _fe_problem.time() = _fe_problem.timeOld();
-
   libmesh_system.update();
 
   // Must compute the residual first
-  _explicit_residual.zero();
-  _fe_problem.computeResidual(*libmesh_system.current_local_solution, _explicit_residual);
-
-  // The residual is on the RHS
+  // compute {R}
+  // {R} = {f} - [K] {u}
+  _fe_problem.computeResidualType(
+      *libmesh_system.current_local_solution, _explicit_residual, _fe_nontime_tag);
   _explicit_residual *= -1.;
 
-  // Compute the mass matrix
-  _fe_problem.computeJacobianTag(*libmesh_system.current_local_solution, mass_matrix, _Ke_time_tag);
-
-  // Still testing whether leaving the old update is a good idea or not
-  // _explicit_euler_update = 0;
-
   auto converged = false;
 
   switch (_solve_type)
   {
     case CONSISTENT:
     {
-      const auto num_its_and_final_tol = _linear_solver->solve(
-          mass_matrix,
-          _explicit_euler_update,
-          _explicit_residual,
+      // get mass matrix [M]
+      _fe_problem.computeJacobianTag(
+          *libmesh_system.current_local_solution, system_matrix, _Ke_second_time_tag);
+
+      // Solve explicit euler correction EEC
+      // rhs = [M] {u_old - u_older}
+      _rhs.zero();
+      _explicit_euler_correction.zero();
+      _explicit_euler_correction += _solution_old;
+      if (_t_step == 1)
+        _explicit_euler_correction -= _solution_minus_one;
+      else
+        _explicit_euler_correction -= _solution_older;
+      system_matrix.vector_mult(_rhs, _explicit_euler_correction);
+      // [M + C] {EEC} = rhs
+      _fe_problem.computeJacobianTag(
+          *libmesh_system.current_local_solution, system_matrix, _Ke_time_tag);
+      _explicit_euler_correction.zero();
+      const auto num_its_and_final_tol_EEC = _linear_solver->solve(
+          system_matrix,
+          _explicit_euler_correction,
+          _rhs,
           es.parameters.get<Real>("linear solver tolerance"),
           es.parameters.get<unsigned int>("linear solver maximum iterations"));
 
       converged = checkLinearConvergence();
+      _n_linear_iterations = num_its_and_final_tol_EEC.first;
+
+      // Solve for explicit euler update EEU
+      // [M + C] {EEU} = {R}
+      _explicit_euler_update.zero();
+      const auto num_its_and_final_tol_EEU = _linear_solver->solve(
```

Comment:
Why are there two linear solves here?  That doesn't make sense... and will be slow.

Reply:
We've completely linearized the problem so I can't understand why there would be two linear solves.

Reply:
I can change it to one linear solve following @cbolisetti 's formulation.